### PR TITLE
Resource Graph: Normalize top level arrays as PS arrays

### PR DESCRIPTION
**Pull request by ilidemi**
## Description

This fix addresses #9225.

When the client prepares the dynamic server response for output, it recursively transforms complex objects to PSCustomObjects. The issue was that the transformation is only invoked for JObjects but not JArrays, so top-level array fields get returned as JArrays. Transformation logic already handles arrays nested in objects so the fix is very simple.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceGraph/ResourceGraph/Az.ResourceGraph.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.7.1'
+ModuleVersion = '0.7.2'
```
Comment (cormacpayne) [inline src/ResourceGraph/ResourceGraph/Az.ResourceGraph.psd1]: @demyanenko you can revert this change -- we'll version this module and publish it during our next release

File: src/ResourceGraph/ResourceGraph/ChangeLog.md [hunk 2]
```diff
@@ -17,4 +17,10 @@
     * Overview of change #1
         - Additional information about change #1
 -->
-## Upcoming Release
\ No newline at end of file
+## Upcoming Release
+
+## Version 0.7.2
```
Comment (cormacpayne) [inline src/ResourceGraph/ResourceGraph/ChangeLog.md]: @demyanenko you can remove this `## Version 0.7.2` header since we'll do the versioning later. The line below will go under the `## Upcoming Release` header. The content you have below with `## Version 0.7.1` and the line below that can stay.

File: src/ResourceGraph/ResourceGraph/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -25,5 +25,5 @@
 [assembly: CLSCompliant(false)]
 [assembly: Guid("cdfd4e1b-bb5f-439b-8843-e6127cd9b8ad")]
 
-[assembly: AssemblyVersion("0.7.1")]
-[assembly: AssemblyFileVersion("0.7.1")]
+[assembly: AssemblyVersion("0.7.2")]
```
Comment (cormacpayne) [inline src/ResourceGraph/ResourceGraph/Properties/AssemblyInfo.cs]: @demyanenko you can revert the changes made to this file since we'll do the versioning later

File: src/ResourceGraph/ResourceGraph/Utilities/TableExtensions.cs [hunk 4]
```diff
@@ -67,7 +67,7 @@ public static IEnumerable<PSObject> ToPsObjects(this Table table)
                 {
                     var value = row[columnIndex];
                     var normalizedValue =
-                        (value as JObject)?.ToPsObject() ?? value;
+                        (value as JToken)?.ToPsObject() ?? value;
```
Comment (cormacpayne) [inline src/ResourceGraph/ResourceGraph/Utilities/TableExtensions.cs]: @demyanenko if we have examples in the markdown files with the corresponding output from running the cmdlets, we should update the output to reflect the changes being made here

Comment (demyanenko) [reply src/ResourceGraph/ResourceGraph/Utilities/TableExtensions.cs]: It is a fairly corner case so the markdown examples are not covering it. I appreciate the concern.
